plugins/org.eclipse.fordiac.ide.debug/src/org/eclipse/fordiac/ide/debug/fb/FBLaunchEventQueue.java

```diff
@@ -79,31 +100,137 @@ public AtomicInteger getCount(final Event ev) {
+	/**
+	 * Set whether to repeat the last event
 	 *
-	 * @param repeat The repeat state */
+	 * @param repeat The repeat state
+	 */
 	public void setRepeat(final boolean repeat) {
 		this.repeat.set(repeat);
 	}
 
-	/** Get whether to block if empty
+	/**
+	 * Get whether to block if empty
 	 *
-	 * @return The blocking state */
+	 * @return The blocking state
+	 */
 	public boolean isBlocking() {
 		return blocking.get();
 	}
 
-	/** Set whether to block if empty
+	/**
+	 * Set whether to block if empty
 	 *
-	 * @param blocking The blocking state */
+	 * @param blocking The blocking state
+	 */
 	public void setBlocking(final boolean blocking) {
 		this.blocking.set(blocking);
 	}
+
+	/**
+	 * Get whether Debug Time is incremented every time an event is triggered.
+	 *
+	 * @return the Debug Time state
+	 */
+	public boolean isDebugTimeIncremental() {
+		return incremental.get();
+	}
+
+	/**
+	 * Set whether Debug Time is incremented every time an event is triggered.
+	 *
+	 * @param incremental the Debug Time state
+	 */
+	public void setDebugTimeIncremental(final boolean incremental) {
+		this.incremental.set(incremental);
+	}
+
+	/**
+	 * Get whether Debug Time is is fixed to manual time value.
+	 *
+	 * @return the Debug Time state
+	 */
+	public boolean isDebugTimeManual() {
+		return manual.get();
+	}
+
+	/**
+	 * Set whether Debug Time is fixed to manual time value.
+	 *
+	 * @param manual the Debug Time state
+	 */
+	public void setDebugTimeManual(final boolean manual) {
+		this.manual.set(manual);
+	}
+
+	/**
+	 * Set whether Debug Time is default system time.
+	 *
+	 * @param system the Debug Time state
+	 */
+	public void setDebugTimeSystem(final boolean system) {
+		this.system.set(system);
+	}
+
+	/**
+	 * Get whether Debug Time is set to default system time (UTC).
+	 *
+	 */
+	public boolean isDebugTimeSystem() {
+		return system.get();
+	}
+
+	/**
+	 * Set the value of the Debug Time. This value is used to set the Executor
+	 * Clock. Depending on the selected mode, this value is either fixed or
+	 * incremented each time an event is triggered.
+	 *
+	 * @param sleepTime the time value
+	 */
+	public void setDebugTimeValue(final Duration sleepTime) {
+		incrementDebugTime = sleepTime;
+		currentDebugTime = incrementDebugTime;
+	}
+
+	/**
+	 * Get the current intervall time.
+	 *
+	 * @return the time as Duration
+	 */
+	public Duration getDebugTimeValue() {
+		return incrementDebugTime;
+	}
+
+	/**
+	 * Set the Evaluator Process.
+	 *
+	 * @param evaluatorProcess the current EvaluatorProcess
+	 */
+	public void setEvaluatorProcess(final EvaluatorProcess evaluatorProcess) {
+		evaluator = evaluatorProcess;
+	}
+
+	/**
+	 * Set Debug Time in Evaluator depending on mode selection.
+	 *
+	 */
+	public void setDebugTime() {
+		final Instant instant = Instant.ofEpochSecond(currentDebugTime.getSeconds(), currentDebugTime.getNano());
+		evaluator.getExecutor().setClock(Clock.fixed(instant, ZoneId.systemDefault()));
```

Comment:
If the evaluator has not been set by the UI (when the *FB Debug* view is not open) this leads to an NPE.